subgraphs/aave-v2/src/mappings/lendingPool.ts

```diff
@@ -0,0 +1,374 @@
+  repay.snapshotId = getDaysSinceEpoch(event.block.timestamp.toI32());
+  repay.save();
+}
+
+export function handleLiquidationCall(event: LiquidationCall): void {
+  // Liquidate event where a users collateral assets are sold to reduce the ratio to borrowed assets
+  const hash = event.transaction.hash.toHexString();
+  const marketAddr = event.params.collateralAsset.toHexString();
+  const protocolId = getProtocolIdFromCtx();
+  const protocol = getOrCreateProtocol(protocolId);
+  const token = getOrCreateToken(event.params.collateralAsset);
+  const liquidate = new LiquidateEntity(hash + '-' + event.logIndex.toHexString());
+  // liquidate.to is set to the market address
+  liquidate.to = marketAddr;
+  liquidate.market = marketAddr;
+  liquidate.from = event.params.liquidator.toHexString();
+  liquidate.hash = hash;
+  liquidate.logIndex = event.logIndex.toI32();
+  liquidate.asset = event.params.collateralAsset.toHexString();
+  liquidate.protocol = protocol.id || protocolId;
+  liquidate.timestamp = event.block.timestamp;
+  liquidate.blockNumber = event.block.number;
+  liquidate.amount = event.params.liquidatedCollateralAmount;
+
+  let market = initMarket(event.block.number, event.block.timestamp, marketAddr) as Market;
+  const amountUSD = amountInUSD(market.inputTokenPricesUSD[0], token.decimals, event.params.liquidatedCollateralAmount, market);
+  liquidate.amountUSD = amountUSD;
+  // The index of the inputToken and the inputTokenBalance are the same, as these arrays push the crresponding values to the same index when added
+  const tokenBalanceIndex = getTokenBalanceIndex(market, liquidate.asset);
+  if (tokenBalanceIndex >= market.inputTokens.length) {
+    // if the getTokenBalanceIndex function added the event asset to the inputTokens array, reinitialize the market
+    market = initMarket(event.block.number, event.block.timestamp, marketAddr) as Market;
+  }
+  const balanceAtIdx = market.inputTokenBalances[<i32>tokenBalanceIndex];
+  const newBal = balanceAtIdx.plus(liquidate.amount);
+  market.inputTokenBalances = setTokenBalanceArray(newBal, tokenBalanceIndex, market.inputTokenBalances);
+
+  // Update total value locked on the market level
+  updateTVL(token, market, protocol, liquidate.amount, false);
+  calculateRevenues(market, token);
+  market.save();
+  // Update snapshots
+  getMarketDailySnapshot(event, market);
+  updateMetricsDailySnapshot(event);
+  updateFinancials(event);
+
+  // Get the surplus percentage of the amount with the liquidation penalty/bonus
+  // The liquidationPenalty is the amount as a percentage of the collateral that was liquidated summed with the percentage of the collateral that is provided as a bonus
+  // This is represented by a 5 digit integer over 10000 (100%). The profit is the percentage over 100%
+  // For example, if the liquidationPenalty is 10500 (105%), the profit is 5% of the collateral liquidated.
```

Comment:
`liquidationPenalty` cannot be above 100%, it's usually 5-10%

Reply:
Reoriented how I save the liquidationPenalty on the market entity. I originally saved it this way because of how it is sent in the event parameter. Added calculations upon receiving this param to format it as a percentage below 100%.

Reply:
Reoriented how I save the liquidationPenalty on the market entity. I originally saved it this way because of how it is sent in the event parameter. Added calculations upon receiving this param to format it as a percentage below 100%.